### PR TITLE
Sync analysis_options.yaml with flutter/flutter

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]
Thanks! So much better.

**Review by zanderso — APPROVED** [review]

**Comment by goderbauer** [comment]
@yjbanov Can you scan through the changes this required in `web_ui`?

**Review by yjbanov — APPROVED** [review]

**Comment by goderbauer** [comment]
@dnfield Would you be up for reviewing the changes to `lib/ui` that this required now that we are properly analyzing it after https://github.com/flutter/engine/pull/34988 got resolved? Those changes are all in the last commit of this PR. All other commits have already been reviewed.

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,77 +1,75 @@
...
+    strict-casts: true
+    strict-raw-types: true
   errors:
     # treat missing required parameters as a warning (not a hint)
     missing_required_param: warning
     # treat missing returns as a warning (not a hint)
     missing_return: warning
-    native_function_body_in_non_sdk_code: ignore
-    # allow having TODOs in the code
+    native_function_body_in_non_sdk_code: ignore # DIFFERENT FROM FLUTTER/FLUTTER
```
Comment (dnfield) [inline analysis_options.yaml:19]: We shouldn't need this anymore now that we're on FfiNative.

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,77 +1,75 @@
...
     # allow dart:ui to import dart:_internal
-    import_internal_library: ignore
+    import_internal_library: ignore # DIFFERENT FROM FLUTTER/FLUTTER
+    # allow self-reference to deprecated members (we do this because otherwise we have
+    # to annotate every member in every test, assert, etc, when we deprecate something)
+    deprecated_member_use_from_same_package: ignore
     # Turned off until null-safe rollout is complete.
     unnecessary_null_comparison: ignore
     # TODO(goderbauer): remove when https://github.com/dart-lang/sdk/issues/49563 is fixed.
     ffi_native_unexpected_number_of_parameters: ignore
```
Comment (dnfield) [inline analysis_options.yaml:30]: This is different from flutter right?

File: analysis_options.yaml (hunk 1 above)
Comment (goderbauer) [reply analysis_options.yaml:19]: Mind if I remove this in a follow-up?

File: analysis_options.yaml (hunk 2 above)
Comment (goderbauer) [reply analysis_options.yaml:30]: Yeah, will leave a comment on it.

File: analysis_options.yaml [hunk 3]
```diff
@@ -1,77 +1,75 @@
...
+    - avoid_print
     # - avoid_private_typedef_functions # we prefer having typedef (discussion in https://github.com/flutter/flutter/pull/16356)
-    # - avoid_redundant_argument_values # not yet tested
+    - avoid_redundant_argument_values
     - avoid_relative_lib_imports
     - avoid_renaming_method_parameters
     - avoid_return_types_on_setters
-    # - avoid_returning_null # there are plenty of valid reasons to return null
-    # - avoid_returning_null_for_future # not yet tested
+    # - avoid_returning_null # still violated by some pre-nnbd code that we haven't yet migrated
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply analysis_options.yaml:68]: This is just syncing over from flutter/flutter and that's what we have there. Happy to take another look at this particular lint in a follow-up.

**Review by dnfield — COMMENTED** [review]
This is overall an improvement, feel free to land and file issues/follow up on my comments subsequently to avoid merge conflicts with bots

File: analysis_options.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnfield) [reply analysis_options.yaml:68]: SGTM, we should be fully NNBD in this repo (except for flutter_frontend_server and a couple stray files in fuchsia embedding.

**Comment by dnfield** [comment]
> @dnfield Would you be up for reviewing the changes to `lib/ui` that this required now that we are properly analyzing it after #34988 got resolved? Those changes are all in the last commit of this PR. All other commits have already been reviewed.

LGTM

File: analysis_options.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (goderbauer) [reply analysis_options.yaml:68]: Trying to turn this on in https://github.com/flutter/engine/pull/35015 to see what (if anything) breaks.

File: analysis_options.yaml (hunk 1 above)
Comment (goderbauer) [reply analysis_options.yaml:19]: Trying to turn this on in https://github.com/flutter/engine/pull/35015 to see what (if anything) breaks.

**Pull request merged**